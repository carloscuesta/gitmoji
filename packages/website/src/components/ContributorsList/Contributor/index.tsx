import styles from './styles.module.css'

type Props = { avatar: string; url: string }

const Contributor = (props: Props) => (
  <article className="col-xs-3 col-sm-2">
    <a href={props.url} target="_blank" rel="noreferrer">
      <img className={styles.picture} src={props.avatar} />
    </a>
  </article>
)

export default Contributor
