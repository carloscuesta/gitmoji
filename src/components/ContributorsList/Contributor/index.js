// @flow
import React, { type Element } from 'react'

type Props = { profile: string, avatar: string }

const Contributor = (props: Props): Element<'article'> => (
  <article className="contributor col-xs-3 col-sm-2">
    <a href={props.profile}>
      <img className="contributor-picture" src={props.avatar} />
    </a>
  </article>
)

export default Contributor
