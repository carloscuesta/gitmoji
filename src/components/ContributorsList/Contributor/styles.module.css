.picture {
  max-width: 100%;
  border-radius: 50%;
  padding: 0.5em;
}
