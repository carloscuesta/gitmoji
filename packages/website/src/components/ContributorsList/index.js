// @flow
import { type Element } from 'react'

import Contributor from './Contributor'

type Props = {
  contributors: Array<{
    avatar: string,
    id: string,
    url: string,
  }>,
}

const ContributorsList = (props: Props): Element<'div'> => (
  <div className="row center-xs">
    {props.contributors.map((contributor) => (
      <Contributor
        key={contributor.id}
        url={contributor.url}
        avatar={contributor.avatar}
      />
    ))}
  </div>
)

export default ContributorsList
