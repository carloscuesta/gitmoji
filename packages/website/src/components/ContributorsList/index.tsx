import Contributor from './Contributor'

type Props = {
  contributors: Array<{
    avatar: string
    id: string
    url: string
  }>
}

const ContributorsList = (props: Props) => (
  <div className="row center-xs">
    {props.contributors.map((contributor) => (
      <Contributor
        key={contributor.id}
        url={contributor.url}
        avatar={contributor.avatar}
      />
    ))}
  </div>
)

export default ContributorsList
