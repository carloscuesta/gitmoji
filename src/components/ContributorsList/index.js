// @flow
import React, { type Element } from 'react'

import Contributor from './Contributor'

const ContributorsList = (): Element<'div'> => {
  const [contributors, setContributors] = React.useState([])

  React.useEffect(() => {
    const fetchContributors = async () => {
      const response = await fetch(
        'https://api.github.com/repos/carloscuesta/gitmoji/contributors'
      )
      const contributors = await response.json()

      setContributors(
        contributors.filter((contributor) => !contributor.login.includes('bot'))
      )
    }

    fetchContributors()
  }, [])

  return (
    <div className="row center-xs">
      {contributors.map((contributor) => (
        <Contributor
          key={contributor.id}
          profile={contributor.html_url}
          avatar={contributor.avatar_url}
        />
      ))}
    </div>
  )
}

export default ContributorsList
