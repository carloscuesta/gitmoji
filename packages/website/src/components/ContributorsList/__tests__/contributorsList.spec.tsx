import renderer from 'react-test-renderer'

import ContributorsList from '../index'
import Contributor from '../Contributor'
import * as stubs from './stubs'

describe('ContributorsList', () => {
  describe('Contributor', () => {
    it('should render the component', () => {
      const wrapper = renderer.create(<Contributor {...stubs.contributor} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  it('should render the component', () => {
    const wrapper = renderer.create(<ContributorsList {...stubs.props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
