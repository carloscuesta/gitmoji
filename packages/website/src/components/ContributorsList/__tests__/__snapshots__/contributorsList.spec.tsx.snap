// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributorsList Contributor should render the component 1`] = `
<article
  className="col-xs-3 col-sm-2"
>
  <a
    href="https://github.com/profile"
    rel="noreferrer"
    target="_blank"
  >
    <img
      className="picture"
      src="https://github.com/avatar"
    />
  </a>
</article>
`;

exports[`ContributorsList should render the component 1`] = `
<div
  className="row center-xs"
>
  <article
    className="col-xs-3 col-sm-2"
  >
    <a
      href="https://github.com/profile"
      rel="noreferrer"
      target="_blank"
    >
      <img
        className="picture"
        src="https://github.com/avatar"
      />
    </a>
  </article>
</div>
`;
